src/profile-logic/process-profile.js

```diff
@@ -1059,53 +1066,193 @@ export function processProfile(
 
 /**
  * Take a processed profile and remove any non-serializable classes such as the
- * StringTable class.
+ *  StringTable class
  */
-export function serializeProfile(
-  profile: Profile,
-  includeNetworkUrls: boolean = true
-): string {
+export function serializeProfile(profile: Profile): string {
   // stringTable -> stringArray
+  const newProfile = Object.assign({}, profile, {
+    threads: profile.threads.map(thread => {
+      const stringArray = thread.stringTable.serializeToArray();
+      // Has to be any since Threads don't have stringArray.
+      const newThread: any = Object.assign({}, thread);
+      delete newThread.stringTable;
+      newThread.stringArray = stringArray;
+      return newThread;
+    }),
+  });
+
+  return JSON.stringify(newProfile);
+}
+
+/**
+ * Take a processed profile with PII that user wants to be removed and remove the
+ * thread data depending on that PII status. Look at `RemoveProfileInformation`
+ * type definition if you want to learn what kind of information we are removing.
+ */
+export function sanitizePII(
+  profile: Profile,
+  PIIToBeRemoved: RemoveProfileInformation
+): Profile {
   let urlCounter = 0;
+  const removedThreadIndexes = [];
   const newProfile = Object.assign({}, profile, {
-    meta: { ...profile.meta, networkURLsRemoved: !includeNetworkUrls },
+    meta: {
+      ...profile.meta,
+      // FIXME: We won't be needing this after PII frontend changes.
+      networkURLsRemoved: PIIToBeRemoved.shouldRemoveNetworkUrls,
+      extensions: PIIToBeRemoved.shouldRemoveExtensions
+        ? getEmptyExtensions()
+        : profile.meta.extensions,
+    },
     pages:
-      includeNetworkUrls === false && profile.pages
+      (PIIToBeRemoved.shouldRemoveNetworkUrls ||
```

Comment:
Please use a ternary rather than the logic switching properties of `&&`. The mix of boolean operators not evaluating to a `boolean` is really confusing here.